### PR TITLE
Add Thanos Rule

### PR DESCRIPTION
**Pull request by kakkoyun**
This PR,
* upgrades dependencies
* adds/modifies manifests to add Thanos Rule

### CODE REVIEW

**Comment by kakkoyun** [comment]
cc @squat

File: environments/openshift/manifests/observatorium-template.yaml [hunk 1]
```diff
@@ -994,6 +1089,8 @@ objects:
           - --whitelist={__name__="node_role_os_version_machine:cpu_capacity_cores:sum"}
           - --whitelist={__name__="node_role_os_version_machine:cpu_capacity_sockets:sum"}
           - --whitelist={__name__="subscription_sync_total"}
+          - --whitelist={__name__="csv_succeeded"}
```
Comment (bwplotka) [inline environments/openshift/manifests/observatorium-template.yaml]: Is this change related?

Comment (kakkoyun) [reply environments/openshift/manifests/observatorium-template.yaml]: This is here because I have also updated dependencies, which comes from https://github.com/openshift/telemeter/pull/253/files. They should've created a PR for here as well.

Should I exclude this change and create another PR for it?

Comment (bwplotka) [reply environments/openshift/manifests/observatorium-template.yaml]: No it's fine, I think thanks for context!

**Comment by metalmatze** [comment]
Is there anything left to do for this one? 😊

**Comment by kakkoyun** [comment]
@metalmatze If you give your blessing, I'll merge it :)

File: environments/kubernetes/manifests/thanos-receive-statefulSet-default.yaml [hunk 2]
```diff
@@ -109,10 +109,6 @@ spec:
           requests:
             cpu: 32m
             memory: 16Mi
-      volumes:
-      - configMap:
-          name: observatorium-tenants-generated
-        name: observatorium-tenants
```
Comment (metalmatze) [inline environments/kubernetes/manifests/thanos-receive-statefulSet-default.yaml]: We still want this, no?

Comment (kakkoyun) [reply environments/kubernetes/manifests/thanos-receive-statefulSet-default.yaml]: I overrode the volume claims that comes from kube-thanos to fix this issue.

**Comment by kakkoyun** [comment]
Rebased after template splits.

**Comment by kakkoyun** [comment]
cc @metalmatze Could you have another look at it?

**Comment by metalmatze** [comment]
The component should probably be named Ruler when we first merge and introduce this already. :blush:

**Comment by kakkoyun** [comment]
Maybe we should do the renaming, as another PR when upstream proposal also happens. Right now, `thanos` and `kube-thanos` refer it as `thanos rule`. At least I have to change it on `kube-thanos` first.

So maybe we can merge this one since this also has pvc/volume fixes in it.

File: environments/kubernetes/manifests/thanos-store-statefulSet.yaml [hunk 3]
```diff
@@ -95,7 +95,6 @@ spec:
           requests:
             cpu: 32m
             memory: 64Mi
-      volumes: null
```
Comment (kakkoyun) [inline environments/kubernetes/manifests/thanos-store-statefulSet.yaml]: @metalmatze Do you think this change would create a problem?

Comment (metalmatze) [reply environments/kubernetes/manifests/thanos-store-statefulSet.yaml]: I don't think so, as this has been null before. But I'm not 100% sure.

Comment (brancz) [reply environments/kubernetes/manifests/thanos-store-statefulSet.yaml]: I think it should be equivalent.

**Comment by kakkoyun** [comment]
@metalmatze To fix volume issues, this disables https://github.com/thanos-io/kube-thanos/pull/76 hiding the volumes.

**Comment by squat** [comment]
I think this was the right move: https://github.com/observatorium/configuration/pull/122#issuecomment-565504394

otherwise it's quite funny to see half thanos-rule and half-thanos-ruler. It seems like a label mismatch just waiting to happen.

**Comment by kakkoyun** [comment]
@metalmatze @squat  Changes in https://github.com/thanos-io/kube-thanos/pull/77 are also reflected now.
